engines/cryomni3d/versailles/engine.cpp

```diff
@@ -0,0 +1,1443 @@
+	_toolbar.init(&_sprites, &_fontManager, &_messages, &_inventory, this);
+
+	_fixedImage = new ZonFixedImage(*this, _inventory, _sprites, &kFixedImageConfiguration);
+
+	// Documentation is needed by noone at init time, let's do it last
+	initDocPeopleRecord();
+	_docManager.init(&_sprites, &_fontManager, &_messages, this);
+
+	initGraphics(640, 480);
+	setMousePos(Common::Point(320, 200));
+
+	_isPlaying = false;
+	_isVisiting = false;
+
+#if !defined(DEBUG_FAST_START) || DEBUG_FAST_START<1
+	playTransitionEndLevel(-2);
+	if (g_engine->shouldQuit()) {
+		return Common::kNoError;
+	}
+	playTransitionEndLevel(-1);
+	if (g_engine->shouldQuit()) {
+		return Common::kNoError;
+	}
+#endif
+
+	bool stopGame = false;
+	while (!stopGame) {
+		bool exitLoop = false;
+		unsigned int nextStep = 0;
+#if defined(DEBUG_FAST_START) && DEBUG_FAST_START>=2
+		nextStep = 27;
+		// Called in options
+		syncOmni3DSettings();
+#endif
+		setCursor(181);
+		while (!exitLoop) {
+			_isPlaying = false;
+			if (!nextStep) {
+				nextStep = displayOptions();
+			}
+			if (nextStep == 40) {
+				// Quit action
+				exitLoop = true;
+				stopGame = true;
+			} else if (nextStep == 27 || nextStep == 28 || nextStep == 65) {
+				// New game, Load game, Next level
+				if (nextStep == 27) {
+					// New game
+#if !defined(DEBUG_FAST_START) || DEBUG_FAST_START<1
+					playTransitionEndLevel(0);
+					if (g_engine->shouldQuit()) {
+						stopGame = true;
+						exitLoop = true;
+						break;
+					}
+#endif
+					changeLevel(1);
+				} else if (nextStep == 28) {
+					// Load game
+					loadGame(_isVisiting, _loadedSave);
+				} else if (nextStep == 65) {
+					changeLevel(_currentLevel + 1);
+				}
+
+				_isPlaying = true;
+				_toolbar.setInventoryEnabled(!_isVisiting);
+				nextStep = 0;
+				_abortCommand = AbortNoAbort;
+
+				gameStep();
+
+				// We shouldn't return from gameStep without an abort command
+				assert(_abortCommand != AbortNoAbort);
+				switch (_abortCommand) {
+				default:
+					error("Invalid abortCommand: %d", _abortCommand);
+					// Shouldn't return
+					return Common::kUnknownError;
+				case AbortFinished:
+				case AbortGameOver:
+					// Go back to menu
+					exitLoop = true;
+					break;
+				case AbortQuit:
+					exitLoop = true;
+					stopGame = true;
+					break;
+				case AbortNewGame:
+					nextStep = 27;
+					break;
+				case AbortLoadGame:
+					nextStep = 28;
+					break;
+				case AbortNextLevel:
+					nextStep = 65;
+					break;
+				}
+			}
+		}
+	}
+	return Common::kNoError;
+}
+
+void CryOmni3DEngine_Versailles::setupFonts() {
+	Common::Array<Common::String> fonts;
+	fonts.push_back("font01.CRF");
+	fonts.push_back("font02.CRF");
+	fonts.push_back("font03.CRF");
+	fonts.push_back("font04.CRF");
+	fonts.push_back("font05.CRF");
+	fonts.push_back("font06.CRF");
+	fonts.push_back("font07.CRF");
+	fonts.push_back("font08.CRF");
+	fonts.push_back("font09.CRF");
+	fonts.push_back("font10.CRF");
+	fonts.push_back("font11.CRF");
```

Comment:
For the macintosh version I had to change this as the fonts are named differently. I don't know if I am supposed to load all the fonts there, but this is what I did.

```
	Common::Array<Common::String> fonts;
	if (getPlatform() == Common::kPlatformMacintosh) {
		fonts.push_back("MPW12.CRF");
		fonts.push_back("MPW9.CRF");
		fonts.push_back("arial11.CRF");
		fonts.push_back("fruitL10.CRF");
		fonts.push_back("fruitL11.CRF");
		fonts.push_back("fruitL12.CRF");
		fonts.push_back("fruitL13.CRF");
		fonts.push_back("fruitL14.CRF");
		fonts.push_back("fruitL16.CRF");
		fonts.push_back("fruitL18.CRF");
		fonts.push_back("fruitL9.CRF");
		fonts.push_back("garamB18.CRF");
		fonts.push_back("garamB20.CRF");
		fonts.push_back("garamB22.CRF");
		fonts.push_back("geneva10.CRF");
		fonts.push_back("geneva12.CRF");
		fonts.push_back("geneva13.CRF");
		fonts.push_back("geneva14.CRF");
		fonts.push_back("geneva15.CRF");
		fonts.push_back("geneva9.CRF");
		fonts.push_back("helvet11.CRF");
		fonts.push_back("helvet12.CRF");
		fonts.push_back("helvet13.CRF");
		fonts.push_back("helvet14.CRF");
		fonts.push_back("helvet16.CRF");
		fonts.push_back("helvet24.CRF");
		fonts.push_back("helvet9.CRF");
	} else {
		fonts.push_back("font01.CRF");
		fonts.push_back("font02.CRF");
		fonts.push_back("font03.CRF");
		fonts.push_back("font04.CRF");
		fonts.push_back("font05.CRF");
		fonts.push_back("font06.CRF");
		fonts.push_back("font07.CRF");
		fonts.push_back("font08.CRF");
		fonts.push_back("font09.CRF");
		fonts.push_back("font10.CRF");
		fonts.push_back("font11.CRF");
	}
```

Reply:
This one may be a little harder than that.
In PC version the fonts are kind of selected by their order. For example, if we switch two push_back, the fonts will be switched on display.
So with your patch, only the first 11 fonts will be used. I am not sure what Cryo did with all these fonts in Mac version.
I am not even sure all 11 fonts are used in PC version.

Reply:
I suspected if would be more complicated that that ;-)
The changed allowed me to at least start the game, but the text display didn't look quite right. I guess I will have to do some digging in the executable...